sql/moz-fx-data-shared-prod/telemetry_derived/feature_usage_v2/script.sql

```diff
@@ -0,0 +1,760 @@
+          COALESCE(addon_id = '@contain-facebook', FALSE)
+        ) AS has_facebook_container_extension,
+        LOGICAL_OR(
+          COALESCE(addon_id = '@testpilot-containers', FALSE)
+        ) AS has_multiaccount_container_extension,
+        LOGICAL_OR(
+          COALESCE(addon_id = 'private-relay@firefox.com', FALSE)
+        ) AS has_private_relay_extension
+      FROM
+        telemetry.addons
+      WHERE
+        submission_date = @submission_date
+        AND sample_id = 0
+        AND normalized_channel = 'release'
+      GROUP BY
+        client_id,
+        submission_date
+    )
+    SELECT
+      *
+    FROM
+      user_type
+    LEFT JOIN
+      main
+    USING
+      (client_id, submission_date)
+    LEFT JOIN
+      events
+    USING
+      (client_id, submission_date)
+    LEFT JOIN
+      addons
+    USING
+      (client_id, submission_date)
+    LEFT JOIN
+      activity_stream_events
+    USING
+      (client_id, submission_date)
+    LEFT JOIN
+      activity_stream_sessions
+    USING
+      (client_id, submission_date)
+  );
+
+-- To ensure backfills replace existing data in the feature usage table,
+-- existing data needs to be deleted explicitly since scripts do not
+-- allow to replace table partitions or writing of results to tables.
+DELETE FROM
+  telemetry_derived.feature_usage_v2
+WHERE
+  submission_date = @submission_date;
+
+-- The following part applies COALESCE to feature metrics and checks for
+-- eligible versions.
+-- The query that is generated and executed here looks like the following:
+--   INSERT INTO telemetry_derived.feature_usage_v2
+--   SELECT
+--     client_id,
+--     ...
+--     IF("78" > f.app_version, NULL, COALESCE(password_filled, CAST(0 AS INT64))) AS password_filled,
+--     ...
+--     IF("52" > f.app_version, NULL, COALESCE(attributed, CAST(0 AS BOOL))) AS attributed,
+--     ...
+--   FROM feature_usages f
+EXECUTE IMMEDIATE(
+  SELECT
+    -- Scripts do not return result data that could be written to a partition with
+    -- our existing machinery, so we have to INSERT the data into the destintion table.
+    'INSERT INTO telemetry_derived.feature_usage_v2 ' || 'SELECT ' || STRING_AGG(
+      IF(
+        `type` IS NULL, -- if no metadata is provided in `version_metadata` take the value as is
+        column,
+        -- otherwise use 'IF("<version>" > f.app_version, NULL, COALESCE(<metric>, CAST(0 AS <type>))) AS <metric>,'
+        -- in query
+        'IF("' || min_version || '" > f.app_version, NULL, COALESCE(' || column || ', CAST(0 AS ' || `type` || '))) AS ' || column
+      )
+    ) || ' FROM feature_usages f'
+  FROM
+    (
+      -- This extracts all the column names (=metrics) from the temporary feature usage table.
+      SELECT
+        REGEXP_EXTRACT_ALL(to_json_string(f), r'"([^"]*)":') AS columns
+      FROM
+        feature_usages f
+      LIMIT
+        1
+    ),
+    UNNEST(columns) column
+  LEFT JOIN
+    version_metadata
+  ON
+    metric = column
+);
```

Comment:
This is where the magic happens to `COALESCE` metric values based on eligible browser version defined in the metadata.